### PR TITLE
fix(types/node): `clearInterval` and `clearTimeout` should accept primitive values

### PR DESCRIPTION
**Pull request by dsherret**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/timers.html#clearintervaltimeout (`Timeout | string | number` for both)
- [x] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~ N/A

### CODE REVIEW

File: types/node/timers.d.ts [hunk 1]
```diff
@@ -77,15 +77,15 @@ declare module 'timers' {
         namespace setInterval {
             const __promisify__: typeof setIntervalPromise;
         }
-        function clearInterval(intervalId: NodeJS.Timeout | undefined): void;
```
Comment (dsherret) [inline types/node/timers.d.ts:80]: I'm not sure why these previously had `| undefined` so I just left that as-is.

File: types/iobroker/iobroker-tests.ts [hunk 2]
```diff
@@ -677,10 +677,6 @@ enumObj.common.members && enumObj.common.members.map(() => 1);
 adapter.clearTimeout(adapter.setTimeout(() => {}, 10));
 adapter.clearInterval(adapter.setInterval(() => {}, 10));
 // $ExpectError
-clearTimeout(adapter.setTimeout(() => {}, 10));
-// $ExpectError
-clearInterval(adapter.setInterval(() => {}, 10));
```
Comment (dsherret) [inline types/iobroker/iobroker-tests.ts:682]: `adapter.setInterval` returns a branded number and so this no longer fails due to the updates.

**Review by mcollina — APPROVED** [review]
lgtm

File: types/iobroker/iobroker-tests.ts (hunk 2 above)
Comment (AlCalzone) [reply types/iobroker/iobroker-tests.ts:682]: Hmm the point of this was that timeouts created with the adapter methods must be removed with the corresponding adapter methods.

I'd like to keep that behavior.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
See comment on iobroker types

File: types/node/timers.d.ts (hunk 1 above)
Comment (eps1lon) [reply types/node/timers.d.ts:80]: For parity with their counterparts in browsers i.e. parity with lib.dom.d.ts

File: types/iobroker/iobroker-tests.ts (hunk 2 above)
Comment (dsherret) [reply types/iobroker/iobroker-tests.ts]: I updated the iobroker types to remove the `number` type.

**Review by AlCalzone — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by andrewbranch — APPROVED** [review]

**Comment by dsherret** [comment]
Ready to merge.

Thanks for reviewing everyone!

**Pull request merged**